lib/competition_api.rb

```diff
@@ -8,49 +8,29 @@ class CompetitionApi < WcaApi
   def self.find(competition_id)
     competition_json = fetch_competition(competition_id)
     CompetitionInfo.new(competition_json)
-  rescue RegistrationError
-    nil
   end
 
   def self.url(competition_id)
     "#{EnvConfig.WCA_HOST}/api/v0/competitions/#{competition_id}"
   end
 
-  def self.find!(competition_id)
-    competition_json = fetch_competition(competition_id)
-    CompetitionInfo.new(competition_json)
-  end
-
   def self.fetch_qualifications(competition_id)
-    Rails.cache.fetch("#{competition_id}/qualifications", expires_in: 5.minutes) do
-      response = HTTParty.get("#{url(competition_id)}/qualifications")
-      case response.code
-      when 200
-        @status = 200
-        response.parsed_response
-      when 404
-        Metrics.registration_competition_api_error_counter.increment
-        raise RegistrationError.new(404, ErrorCodes::COMPETITION_NOT_FOUND)
-      else
-        Metrics.registration_competition_api_error_counter.increment
-        raise RegistrationError.new(response.code.to_i, ErrorCodes::COMPETITION_API_5XX)
-      end
+    self.get_request("#{url(competition_id)}/qualifications")
+  rescue RegistrationError => e
+    if (e.data[:http_code] = 404)
```

Comment:
I hope you're aware that in Ruby, comparisons are done using `==`?

In its current state, the `if` block reads "If you successfully *assigned* value 404 to the data, then proceed". In reality, that write operation will virtually never fail, and horrible things will happen in production.

Reply:
Lol yes I am well aware, copy-pasted typo. But it does illuminate the lack of tests on a 404 error, so I'll add those